mesonbuild/compilers.py

```diff
@@ -2729,3 +2733,135 @@ def unix_compile_flags_to_native(self, args):
+        # Don't know how icc works on OSX
+        #if self.icc_type == ICC_OSX:
+        #    return ['-bundle']
+        return ['-shared']
+
+class IntelCCompiler(IntelCompiler, CCompiler):
+    def __init__(self, exelist, version, icc_type, is_cross, exe_wrapper=None):
+        CCompiler.__init__(self, exelist, version, is_cross, exe_wrapper)
+        IntelCompiler.__init__(self, icc_type)
+        self.warn_args = {'1': ['-Wall', '-w3', '-diag-disable:remark', '-Wpch-messages'],
+                          '2': ['-Wall', '-w3', '-diag-disable:remark', '-Wextra-tokens', '-Wpch-messages'],
+                          '3' : ['-Wall', '-w3', '-diag-disable:remark', '-Wpedantic', '-Wextra', '-Wpch-messages']}
+
+    def get_options(self):
+        opts = {'c_std' : coredata.UserComboOption('c_std', 'C language standard to use',
+                                                   ['none', 'c89', 'c99', 
+                                                    'gnu89', 'gnu99'],
+                                                   'none')}
+        return opts
+
+    def get_option_compile_args(self, options):
+        args = []
+        std = options['c_std']
+        if std.value != 'none':
+            args.append('-std=' + std.value)
+        return args
+
+    def get_std_shared_lib_link_args(self):
+        return ['-shared']
+
+
+class IntelCPPCompiler(IntelCompiler, CPPCompiler):
+
+    def __init__(self, exelist, version, icc_type, is_cross, exe_wrap):
+        CPPCompiler.__init__(self, exelist, version, is_cross, exe_wrap)
+        IntelCompiler.__init__(self, icc_type)
+        self.warn_args = {'1': ['-Wall', '-w3', '-diag-disable:remark', '-Wpch-messages', '-Wnon-virtual-dtor'],
+                          '2': ['-Wall', '-w3', '-diag-disable:remark', '-Wextra', '-Wpch-messages', '-Wnon-virtual-dtor'],
+                          '3': ['-Wall', '-w3', '-diag-disable:remark', '-Wpedantic', '-Wextra', '-Wpch-messages', '-Wnon-virtual-dtor']}
+
+    def get_options(self):
+        if mesonlib.version_compare(self.version, '>=17.0.0'):
+            opts = {'cpp_std' : coredata.UserComboOption('cpp_std', 'C++ language standard to use',
+                                                         ['none', 'c++11', 'c++0x', 'c++14',
+                                                           'gnu++98', 'gnu++11', 'gnu++0x', 'gnu++14'],
+                                                         'none'),
+                    'cpp_debugstl': coredata.UserBooleanOption('cpp_debugstl',
+                                                               'STL debug mode',
+                                                               False)}
+        else:
+            opts = {'cpp_std' : coredata.UserComboOption('cpp_std', 'C++ language standard to use',
+                                                         ['none', 'c++03', 'c++11', 'c++0x', 
+                                                           'gnu++98', 'gnu++03', 'gnu++11', 'gnu++0x'],
```

Comment:
__[flake8]__
* [E127] continuation line over-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->